cross-dom-comm/README.md

```diff
@@ -0,0 +1,237 @@
+[You can read more details about this process here](https://community.optimism.io/docs/developers/bridge/messaging/).
+
+## Seeing it in action
+
+To show how this works we installed [a slightly modified version of HardHat's `Greeter.sol`](contracts/Greeter.sol) on both L1 Kovan and Optimistic Kovan
+
+
+| Network | Greeter address  |
+| ------- | ---------------- |
+| Kovan (L1) | [0x11fB328D5Bd8E27917535b6d40b881d35BC39Be0](https://kovan.etherscan.io/address/0x11fB328D5Bd8E27917535b6d40b881d35BC39Be0) |
+| Optimistic Kovan (L2) | [0xD4c204223d6F1Dfad0b7a0b05BB0bCaB6665e0c9](https://kovan-optimistic.etherscan.io/address/0xD4c204223d6F1Dfad0b7a0b05BB0bCaB6665e0c9) |
+
+
+### Setup
+
+This setup assumes you already have [Node.js](https://nodejs.org/en/) and [yarn](https://classic.yarnpkg.com/) installed on your system. 
+
+1. Edit `hardhat.config.js`:
+   1. Set `mnemonic` to point to an account that has ETH on the Kovan test network. 
+   1. Set `module.exports.networks.kovan` to point to a URL that accesses the Kovan test network.
+
+1. Install the necessary packages.
+
+   ```sh
+   yarn
+   ```
+
+### L1 message to L2
+
+1. [Browse to the Greeter contract on Etherscan](https://kovan-optimistic.etherscan.io/address/0xD4c204223d6F1Dfad0b7a0b05BB0bCaB6665e0c9#readContract) and click **greet** to see the greeting.
+
+1. Connect the Hardhat console to Kovan (L1):
+
+   ```sh
+   yarn hardhat console --network kovan
+   ```
+
+1. Deploy and call the `ControlL2Greeter` contract.
+
+   ```js
+   Controller = await ethers.getContractFactory("ControlL2Greeter")
+   controller = await Controller.deploy()
+   tx = await controller.setGreeting("Shalom")
+   rcpt = await tx.wait()
+   ```
+
+1. Make a note of the address of `ControlL2Greeter`.
+
+   ```js
+   controller.address
+   ```
+
+1. [Browse to the Greeter contract on Etherscan](https://kovan-optimistic.etherscan.io/address/0xD4c204223d6F1Dfad0b7a0b05BB0bCaB6665e0c9#readContract) and click **greet** to see the new greeting.
+
+
+
+### L2 message to L1
+
+
+1. [Browse to the Greeter contract on Etherscan](https://kovan.etherscan.io/address/0x11fB328D5Bd8E27917535b6d40b881d35BC39Be0#readContract) and click **greet** to see the greeting.
+
+1. Connect the Hardhat console to Optimistic Kovan (L2):
+
+   ```sh
+   yarn hardhat console --network kovan
+   ```
+
+1. Deploy and call the `ControlL1Greeter` contract.
+
+   ```js
+   Controller = await ethers.getContractFactory("ControlL1Greeter")
+   controller = await Controller.deploy()
+   tx = await controller.setGreeting("Shalom")
+   rcpt = await tx.wait()
+   ```
+
+1. Make a note of the address of `ControlL1Greeter`.
+
+   ```js
+   controller.address
+   ```
+
+1. Keep a copy of the transaction hash.
+
+   ```js
+   tx.hash
+   ```
+
+Once the fault challenge period is over (one minute on Kovan, seven days on the production network) it is necessary to claim the transaction on L1. 
```

Comment:
I think you need to explain what the challenge period is (at least a quick summary) and then link out to the documentation.